src/types/props.ts

```diff
@@ -0,0 +1,35 @@
+import { Color } from "./color";
+import { Font } from "./font";
+import { MacroMap } from "../redux/csState";
+import { WidgetActions } from "../ui/widgets/widgetActions";
+import { Border } from "./border";
+
+export type GenericProp =
+  | string
+  | boolean
+  | number
+  | Color
+  | Font
+  | Border
+  | Rule[]
+  | MacroMap
+  | WidgetActions;
+
+interface Expression {
+  boolExp: string;
+  value: string;
+  convertedValue?: GenericProp;
+}
+
+interface PV {
```

Comment:
Could we give this a more descriptive name? It appears to be only used for `Rules` below but could still cause confusion later

Reply:
Makes sense.